lib/components/CardCarousel/CardCarousel.tsx

```diff
@@ -0,0 +1,127 @@
+  // which is required to allow calling the scrollTo methods on the carousel.
+  const carousel = useCarousel();
+  const rootRef = React.createRef<HTMLDivElement>();
+  const CARD_WIDTH = 260; // This needs to match the width of the MediaCard component
+  const GAP = 16;
+  const [cardsPerSlide, setCardsPerSlide] = React.useState(1);
+  const [pageNr, setPageNr] = React.useState(0);
+  const [pageCount, setPageCount] = React.useState(0);
+
+  const goToNextPage = () => {
+    if (pageNr + 1 < Math.ceil(items.length / cardsPerSlide)) {
+      setPageNr(pageNr + 1);
+      carousel.scrollTo((pageNr + 1) * cardsPerSlide);
+    }
+  };
+  const goToPreviousPage = () => {
+    if (pageNr - 1 >= 0) {
+      setPageNr(pageNr - 1);
+      carousel.scrollTo((pageNr - 1) * cardsPerSlide);
+    }
+  };
+  const goToPage = (page: number) => () => {
+    setPageNr(page);
+    carousel.scrollTo(page * cardsPerSlide);
+  };
+
+  React.useEffect(() => {
+    const current = rootRef.current;
+    const handleResize = (entries: ResizeObserverEntry[]) => {
+      for (const entry of entries) {
+        if (entry.target == current) {
+          setCardsPerSlide(Math.max(Math.floor(entry.contentRect.width / (CARD_WIDTH + GAP)), 1));
+          setPageCount(Math.ceil(items.length / cardsPerSlide));
+          setPageNr(Math.min(Math.max(0, pageNr), Math.ceil(items.length / cardsPerSlide) - 1));
+          carousel.scrollTo(pageNr * cardsPerSlide);
+        }
+      }
+    };
+
+    const resizeObserver = new ResizeObserver(handleResize);
+    if (current) {
+      resizeObserver.observe(current);
+    }
+
+    return () => {
+      if (current) {
+        resizeObserver.unobserve(current);
+      }
+    };
+  }, [cardsPerSlide, carousel, items.length, pageNr, rootRef]);
+
+  // Since carousel RootProvider is being used, setting the gap and slide size via props is not possible.
+  // Instead, we need to set the CSS variables directly on the RootProvider element.
+  const carouselCssVariables = {
+    '--slide-spacing': `${GAP}px`,
+    '--slide-size': `${CARD_WIDTH}px`,
+  } as Readonly<React.CSSProperties>;
+
+  return (
+    <ArkCarousel.RootProvider value={carousel} style={carouselCssVariables}>
+      <ArkCarousel.Viewport className="ds-overflow-hidden" ref={rootRef}>
+        <ArkCarousel.ItemGroup>
+          {items.map((item, index) => {
+            // Change the page according to focused item
+            const handleFocus = () => {
+              const pageWhereFocusedItemIs = Math.floor(index / cardsPerSlide);
+              const focusedItemIsOutsideCurrentPage = pageWhereFocusedItemIs !== pageNr;
+              if (focusedItemIsOutsideCurrentPage) {
+                setPageNr(pageWhereFocusedItemIs);
+                goToPage(pageWhereFocusedItemIs)();
+              }
+            };
+            return (
+              // eslint-disable-next-line react/no-array-index-key
+              <ArkCarousel.Item key={index} index={index} asChild>
```

Comment:
Now that I'm looking at this, I think that when creating the items list to be passed as props, the card label or some other identifier could be set as the id so that index doesn't need to be used here.